### PR TITLE
[PORT] 515 Compatibility

### PR DESCRIPTION
**Pull request by QuacksQ**
## About The Pull Request
Moves us to 515, Probably? I honestly dont know what I'm doing and I'm in over my head on this one so I'm just praying it works.

(Same PR as #152 because I was dumb and deleted my repo lol)

Ports the following from TG:
- https://github.com/tgstation/tgstation/pull/72033
- https://github.com/tgstation/tgstation/pull/72401
- https://github.com/tgstation/tgstation/pull/72657
- https://github.com/tgstation/tgstation/pull/73788
- https://github.com/tgstation/tgstation/pull/74380
## How Does This Help ***Gameplay***?
tiny Preformance increases so slightly smoother gaming.

## How Does This Help ***Roleplay***?
Minimal impact on roleplay.

## Proof of Testing

I really dont know how to test this so just trust me and CI.

image
## Changelog

:cl:
code: Moves us to completely to 515.
/:cl:

### CODE REVIEW

**Comment by RimiNosha** [comment]
Reminder that the current CI fail is caused by viewport, @nevimer

**Comment by nevimer** [comment]
without the winset the verb effectively does nothing, I think, other than update a preference

**Comment by RimiNosha** [comment]
Preferences that don't exist, cause the interface is for clientless mobs, so we're good there.

**Comment by nevimer** [comment]
> Preferences that don't exist in that case, cause the interface is for clientless mobs, so we're good there.

No, you need this winset otherwise you can't change your pixel size at all. CI Will be happy if it has nothing to run...

File: code/modules/client/verbs/viewport.dm [hunk 1]
```diff
@@ -5,7 +5,7 @@ GLOBAL_LIST_INIT(valid_icon_sizes, list(32, 48, 64, 96, 128))
 /// General handler for viewport updates. Takes an argument for an icon size, if not aligning with your preference, updates your preference.
 /client/verb/SetWindowIconSize(val as num|text)
 	set hidden = 1
-	if(!istype(src, /client)) // This can be called on the client interface datum for whatever reason.
+	if(istype(src, /client)) // This can be called on the client interface datum for whatever reason.
 		winset(src, "mapwindow.map", "icon-size=[val]")
```
Comment (nevimer) [inline code/modules/client/verbs/viewport.dm:9]: This should just be on line 11, the point is to update the view and save it, not just update the view and do nothing. Aaaaaaaaaa.

Comment (RimiNosha) [reply code/modules/client/verbs/viewport.dm:9]: I am a smart coder.

**Comment by RimiNosha** [comment]
If this breaks other PRs, we'll just make them work<sup>:tm:</sup>

**Pull request merged**